pkg/nodeagent/controller/operatingsystemconfig/reconciler.go

```diff
@@ -468,3 +470,7 @@ func (r *Reconciler) executeUnitCommands(ctx context.Context, log logr.Logger, n
 
 	return mustRestartGardenerNodeAgent, flow.Parallel(fns...)(ctx)
 }
+
+func (r *Reconciler) afero() afero.Afero {
```

Comment:
Why do we need this function? 😕

Reply:
Yes, I'm rather wondering why we have to call `r.afero().WriteFile` now instead of `r.FS.WriteFile`. Shouldn't this be still possible? `r.FS` if of type `filespkg.NodeAgentFileSystem` which has a `WriteFile` method?

Reply:
OK, I get it now: `NodeAgentFileSystem` just embeds `afero.Fs`, not `afero.Afero`. OK, anyways, we can make it work.